### ISSUE TITLE
error 500 creating a note

### ISSUE THREAD

**Issue by smichea** [SkymindIO/SKIL-CE#58]
following the get started tutorial on SKIL latest (Aug 13 2019), i managed to create a workspace but when creating a first notebook i get an error 500. 
I am using the docker image on win10. I managed to open zeppelin on port 8080 and create a notebook inside

![image](https://user-images.githubusercontent.com/16659140/62947842-d34adc00-be15-11e9-8313-59c4969d96d4.png)

**Comment by smichea** [comment]
in the logs of docker i have :

grep: /var/log/skil/zeppelin-*.log: No such file or directory
Waiting for interpreter install...

but i see the interpreter process in skill gui :
![image](https://user-images.githubusercontent.com/16659140/62948088-566c3200-be16-11e9-8cdb-14e322db3207.png)

**Comment by ShamsUlAzeem** [comment]
@smichea can you share the log files with us? Maybe create a volume for the location /var/log/skil folder and zip it.

**Comment by smichea** [comment]
@ShamsUlAzeem In fact creating this volume allowed zeppelin to create its log files (look like he was missing the rights in the var/log/skill dir of the container).
After that i can correctly create notebook.

Thank you

**Comment by ShamsUlAzeem** [comment]
Good to hear :). We'll probably remove the ubuntu related images in the future and just keep the centos ones. Those ones doesn't have this permission issue for zeppelin.

**Comment by ShamsUlAzeem** [comment]
@smichea can you share the exact image (and tag) that you used for this experiment (from docker)

**Comment by smichea** [comment]
this is what i see inspecting the image :

 "Id": "sha256:9172bbb57f88b3d7d204cca2cb83840385c114c1485c8fa8ebeb4c6b0550f941",
        "RepoTags": [
            "skymind/skil:latest"
        ],
        "RepoDigests": [
            "skymind/skil@sha256:cc698c5d2f4653861763c6d3baca356c737c91d4131f44de15b3f2f462f619b1"
        ],
        "Parent": "",
        "Comment": "",
        "Created": "2019-04-08T17:37:03.734881115Z",

is it sufficient ?

**Comment by ShamsUlAzeem** [comment]
Sure that's enough. Btw you showed 2 images here: 
`skymind/skil:latest`
and 
`skymindops/skil-ce:latest`

the later ones are the older images. I prefer using `skymind/skil:latest`

**Comment by wmeddie** [comment]
docker's handling of :latest can be strange, always prefer using the version number explicitly to avoid getting surprised by a cached and old :latest.

**Comment by ShamsUlAzeem** [comment]
Btw you can try out snapshot skil docker images that has some of the bugs fixed in them. 
`docker run -it -p 9008:9008 -p 8080:8080 skymind/skil-snapshots:1.2.2-zepp-exp`

**Comment by smichea** [comment]
using skil-snaphots:1.2.2-zepp

I get a 401 invalid authorization tocken:/services when creating a workspace (with no tag). 
Here are the log of the Exception in daemon.log :

```
22:49:00.677 [application-akka.actor.default-dispatcher-11] ERROR io.skymind.auth.JWTUtil - The Token's Signature result
ed invalid when verified using the Algorithm: HmacSHA256
22:49:00.677 [application-akka.actor.default-dispatcher-13] ERROR io.skymind.auth.JWTUtil - The Token's Signature result
ed invalid when verified using the Algorithm: HmacSHA256
22:49:00.677 [application-akka.actor.default-dispatcher-11] ERROR io.skymind.auth.JWTUtil - Invalid authorization token:
 /services
22:49:00.677 [application-akka.actor.default-dispatcher-13] ERROR io.skymind.auth.JWTUtil - Invalid authorization token:
 /workspaces
22:49:00.677 [application-akka.actor.default-dispatcher-13] ERROR io.skymind.auth.FunctionUtil - Failed auth Check: {Hos
t=[Ljava.lang.String;@4cce9aef, Accept=[Ljava.lang.String;@236a8366, Referer=[Ljava.lang.String;@452bd4f0, Connection=[L
java.lang.String;@673cde65, User-Agent=[Ljava.lang.String;@5f9d398f, Authorization=[Ljava.lang.String;@281bf0a9, Accept-
Encoding=[Ljava.lang.String;@330c3d0e, Accept-Language=[Ljava.lang.String;@32c202fc}, Invalid authorization token: /work
spaces
22:49:00.677 [application-akka.actor.default-dispatcher-11] ERROR io.skymind.auth.FunctionUtil - Failed auth Check: {Hos
t=[Ljava.lang.String;@44bc3a99, Accept=[Ljava.lang.String;@5a3ffb9c, Referer=[Ljava.lang.String;@26059f5d, Connection=[L
java.lang.String;@3c140641, User-Agent=[Ljava.lang.String;@19598ba8, Authorization=[Ljava.lang.String;@3c9f6004, Accept-
Encoding=[Ljava.lang.String;@36eaffbd, Accept-Language=[Ljava.lang.String;@2ad0d12f}, Invalid authorization token: /serv
ices
22:49:00.678 [application-akka.actor.default-dispatcher-13] ERROR io.skymind.auth.FunctionUtil - Failed JWT Check: /work
spaces, Invalid authorization token: /workspaces
22:49:00.678 [application-akka.actor.default-dispatcher-11] ERROR io.skymind.auth.FunctionUtil - Failed JWT Check: /serv
ices, Invalid authorization token: /services
22:49:00.678 [application-akka.actor.default-dispatcher-13] WARN  io.skymind.skil.play.Results - Error [GET] /workspaces
java.lang.Exception: Invalid authorization token: /workspaces
        at io.skymind.auth.JWTUtil.checkJWT(JWTUtil.java:143)
        at io.skymind.auth.FunctionUtil.checkAuth(FunctionUtil.java:244)
        at io.skymind.auth.FunctionUtil.access$000(FunctionUtil.java:29)
        at io.skymind.auth.FunctionUtil$1.get(FunctionUtil.java:74)
        at io.skymind.auth.FunctionUtil$1.get(FunctionUtil.java:61)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at play.routing.RouterBuilderHelper$$anonfun$build$1$$anonfun$apply$1$$anonfun$5.apply(RouterBuilderHelper.scala
:66)
        at play.routing.RouterBuilderHelper$$anonfun$build$1$$anonfun$apply$1$$anonfun$5.apply(RouterBuilderHelper.scala
:62)
        at play.api.mvc.Action$.invokeBlock(Action.scala:533)
        at play.api.mvc.Action$.invokeBlock(Action.scala:530)
        at play.api.mvc.ActionBuilder$$anon$1.apply(Action.scala:493)
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:105)
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:105)
        at play.utils.Threads$.withContextClassLoader(Threads.scala:21)
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:104)
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:103)
        at scala.Option.map(Option.scala:145)
        at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:103)
        at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:96)
        at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:524)
        at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:524)
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:560)
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:560)
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$13.apply(Iteratee.scala:536)
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$13.apply(Iteratee.scala:536)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
22:49:00.678 [application-akka.actor.default-dispatcher-11] WARN  io.skymind.skil.play.Results - Error [GET] /services
java.lang.Exception: Invalid authorization token: /services
        at io.skymind.auth.JWTUtil.checkJWT(JWTUtil.java:143)
        at io.skymind.auth.FunctionUtil.checkAuth(FunctionUtil.java:244)
        at io.skymind.auth.FunctionUtil.access$000(FunctionUtil.java:29)
        at io.skymind.auth.FunctionUtil$1.get(FunctionUtil.java:74)
        at io.skymind.auth.FunctionUtil$1.get(FunctionUtil.java:61)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at play.routing.RouterBuilderHelper$$anonfun$build$1$$anonfun$apply$1$$anonfun$5.apply(RouterBuilderHelper.scala
:66)
        at play.routing.RouterBuilderHelper$$anonfun$build$1$$anonfun$apply$1$$anonfun$5.apply(RouterBuilderHelper.scala
:62)
        at play.api.mvc.Action$.invokeBlock(Action.scala:533)
        at play.api.mvc.Action$.invokeBlock(Action.scala:530)
        at play.api.mvc.ActionBuilder$$anon$1.apply(Action.scala:493)
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:105)
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:105)
        at play.utils.Threads$.withContextClassLoader(Threads.scala:21)
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:104)
        at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:103)
        at scala.Option.map(Option.scala:145)
        at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:103)
        at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:96)
        at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:524)
        at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:524)
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:560)
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:560)
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$13.apply(Iteratee.scala:536)
        at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$13.apply(Iteratee.scala:536)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)

```
beside this there are no error now creating a note, thx!

**Comment by wmeddie** [comment]
Thanks for the issue report. This error is caused from a cached token from a previous session being used to check if the token is valid. It should be benign. But we will fix it before the final 1.2.2 release.
